tools/checkrelease.sh

```diff
@@ -19,114 +19,218 @@
+      echo " OK: $RELEASE_FILE sha512 hash matches."
+    else
+      RETURN_CODE=1
+      echo " - $RELEASE_FILE sha512 hash does not match:"
+      echo "$output"
+    fi
+    echo
 }
 
 function check_gpg() {
-    # check nuttx sha512 and gpg
     RELEASE_FILE=$1
     echo "Checking $RELEASE_FILE GPG signature:"
-    gpg --verify "$RELEASE_FILE.asc" "$RELEASE_FILE"
+    output="$(gpg --quiet --verify $RELEASE_FILE.asc $RELEASE_FILE 2>&1)"
+    return_value=$?
+    if [ $return_value -eq 0 ]; then
+      echo " OK: $RELEASE_FILE gpg signature matches."
+    else
+      RETURN_CODE=1
+      echo " - Error checking $RELEASE_FILE gpg signature:"
+      echo "$output"
+      echo
+    fi
     echo
 }
 
 function check_required_files() {
-    # check nuttx for required files
-    RELEASE_FILE=$1
-    RELEASE_DIR=$2 
-    rm -rf "$RELEASE_DIR"
-    tar xf "$RELEASE_FILE"
+    RELEASE_DIR=$1
+    echo "Checking $RELEASE_FILE for required files:"
     ERROR=0
     if [ ! -f "$RELEASE_DIR/LICENSE" ]; then
-        echo "LICENSE file not present."
-        ERROR=1
+      echo " - LICENSE file not present."
+      ERROR=1
     fi
     if [ ! -f "$RELEASE_DIR/NOTICE" ]; then
-        echo "NOTICE file not present."
-        ERROR=1
+      echo " - NOTICE file not present."
+      ERROR=1
     fi
     if [ ! -f "$RELEASE_DIR/README.txt" ]; then
-        echo "README.txt file not present."
-        ERROR=1
+      echo " - README.txt file not present."
+      ERROR=1
     fi
     if [ ! -f "$RELEASE_DIR/DISCLAIMER-WIP" ]; then
-        echo "DISCLAIMER-WIP file not present."
-        ERROR=1
+      echo " - DISCLAIMER-WIP file not present."
+      ERROR=1
     fi
     if [ 0 -eq $ERROR ]; then
-        echo "OK: All required files exist."
+      echo " OK: all required files exist in $RELEASE_DIR."
+    else
+      RETURN_CODE=1
     fi
+    echo
 }
 
 function check_nuttx() {
-    # check nuttx sha512 and gpg
     RELEASE_FILE="$(ls *.tar.gz|head -1)"
-    check_sha512 "$RELEASE_FILE" 
+    RELEASE_DIR="nuttx"
+    check_sha512 "$RELEASE_FILE"
     check_gpg "$RELEASE_FILE"
-    check_required_files "$RELEASE_FILE" "nuttx"
-    mv "$RELEASE_FILE" ..
+    rm -rf "$RELEASE_DIR"
+    tar xf "$RELEASE_FILE"
+    check_required_files "$RELEASE_DIR"
+    mv "$RELEASE_FILE" "$TMP"
 }
 
 function check_nuttx_apps() {
-    # check nuttx-apps sha512 and gpg
     RELEASE_FILE="$(ls *.tar.gz|head -2| tail -1)"
+    RELEASE_DIR="apps"
     check_sha512 "$RELEASE_FILE"
     check_gpg "$RELEASE_FILE"
-    check_required_files "$RELEASE_FILE" "apps"
-    mv "$RELEASE_FILE" ..
+    rm -rf "$RELEASE_DIR"
+    tar xf "$RELEASE_FILE"
+    check_required_files "$RELEASE_DIR"
+    mv "$RELEASE_FILE" "$TMP"
 }
 
+function check_sim_nsh_build() {
+    RELEASE_DIR="nuttx"
+    echo "Trying to build $RELEASE_DIR sim:nsh..."
+    cd "$RELEASE_DIR"
+    if [[ $VERBOSE -eq 0 ]]; then
+      output=$(make distclean ; ./tools/configure.sh sim:nsh ; make) 2>&1
```

Comment:
Move 2>&1 into ()

Reply:
@xiaoxiang781216 If I do that it, I think it will only apply to the last command, not all of them. I don't think that's what you want— is it?

Reply:
If so, how about we change (2>&1) out $(gpg ...) too?

Reply:
@xiaoxiang781216 I thought more about @btashton's comment above, and removed the option to quiet the GPG invocation completely. He suggested seeing the output since there's info there that would be lost with a simple "Ok". So I'm having the GPG output show on every run.